src/openforms/submissions/models/submission.py

```diff
@@ -642,3 +648,31 @@ def get_prefilled_data(self):
                 variable.key: variable.value for variable in prefill_vars
             }
         return self._prefilled_data
+
+    def get_cosigner_email(self) -> str | None:
+        for form_step in self.form.formstep_set.select_related("form_definition"):
+            for (
+                component,
+                upload_info,
+                data_path,
+                configuration_path,
+            ) in iterate_data_with_components(
+                form_step.form_definition.configuration, self.data
+            ):
+                if component["type"] == "cosign":
+                    return glom(self.data, data_path, default=None)
+
+    @property
+    def waiting_on_cosign(self) -> bool:
+        if self.cosign_complete:
+            return False
+
+        # Cosign not complete, but required
+        if self.form.cosigning_required:
+            return True
+
+        # Co-sign not complete, not required, but the optional cosign component was filled in with an email
+        if self.get_cosigner_email():
+            return True
+
+        return False
```

Comment:
Doesn't this read like English already?

```suggestion
    def waiting_on_cosign(self) -> bool:
        return (not self.cosign_complete 
            and self.form.cosigning_required
            or self.get_cosigner_email() != ""
        )
```

We are waiting when we're not done and it's required or we have an email set.

Reply:
Heh I always struggle to parse in my head multiple bool conditions at once, so I tend to spread them out :sweat_smile:  I can make it a bit more compact.